### PR TITLE
393 - Text Annotation Font Options

### LINKED ISSUE #393: Users can't create a new zoning district inside of a larger one
Users will frequently want to draw a new zoning district inside of a larger one that will surround it on all sides. Essential they want to carve a donut hole inside an larger district, and create a new district in the hole. Currently, this is not possible.

Options to resolve this:
- ~Automatically cut out and delete the overlap somehow~
- **Allow users to have overlapping layers but have them add free form labels for the exterior district so it can be discerned from the inside one.**

### PR DESCRIPTION
**Pull request by andycochran**
This PR will address #393 by allowing users to adjust the font size and weight of text annotations. 

I've stubbed in this UI: 

image

Need to: 
- [ ] add actions to "Large" button and "Bold" button that toggle font state
- [ ] replace hardcoded booleans with a real conditions to indicate state of buttons
- [ ] save `large` and `bold` to the label
- [ ] use `large` and `bold` to style the layer(s) accordingly
- [ ] make sure two sizes and weights of text labels match polygon styles

@Taylor, you were doing some work on styling labels. Would you like to pair on finishing this?

closes #393

### CODE REVIEW

**Comment by andycochran** [comment]
Currently, these "Large" and "Bold" buttons show up for text labels AND zoning polygons. Need to: 
- [ ] Either 1) conditionally show large/bold options only for text labels OR 2) have altered zoning labels default to large/bold with the option of changing

**Comment by allthesignals** [comment]
@andycochran @julialucyhogan we also have this.

I'm not sure how I feel about using `[WIP]` for PRs. This gets pushed to `In Progress` but sometimes forgotten because there's a separate handoff step.

File: app/components/project-geometries/modes/draw/feature-label-form.js [hunk 1]
```diff
@@ -18,13 +18,50 @@ export default class FeatureLabelFormComponent extends Component {
...
 
+  isBold = false;
+
+  isLarge = false;
+
+  @observes('selectedFeature')
+  selectedFeatureUpdated() {
+    this.set('isBold', false);
+    this.set('isLarge', false);
+  }
```
Comment (allthesignals) [inline app/components/project-geometries/modes/draw/feature-label-form.js]: Not sure that observes is the right choice here. `isBold` and `isLarge` are conceptually able to be computed downstream of ... `selectedFeature` being mutated? What about making `isBold` and `isLarge` into computeds themselves? 

There is also `this.notifyPropertyChange`. 

Lastly, we have a `updateSelectedFeature` - what if `updateSelectedFeature` was kept as a public argument, but we create a private method which internally calls `updateSelectedFeature` while also triggering necessary updates?

Or we could just keep an observer in the interest of time.

File: app/components/project-geometries/modes/draw/feature-label-form.js [hunk 2]
```diff
@@ -18,13 +18,50 @@ export default class FeatureLabelFormComponent extends Component {
...
+  geomType;
+
   @action
   handleSelectChange(newLabel) {
     this.set('label', newLabel);
-    this.updateSelectedFeature(newLabel);
+    this.updateSelectedFeature('label', newLabel);
+    if (this.get('geomType') === 'Polygon') {
+      this.updateSelectedFeature('textFont', 'bold');
+    }
```
Comment (allthesignals) [inline app/components/project-geometries/modes/draw/feature-label-form.js]: This is starting to reveal the need for subclassing of some sort (https://sourcemaking.com/refactoring/replace-type-code-with-subclasses). We have all this geometry type-specific handled for a feature-label-form. Maybe geometry-types are _labelable_ things and get access to some service for that? Anyways, for another time...

**Pull request merged**